### PR TITLE
Update procedure for Che v7.41 and earlier

### PR DESCRIPTION
**Pull request by l0rd**
## What does this pull request change

Replaces #2159

## What issues does this pull request fix or reference

eclipse/che/issues/20614

## Pull request checklist

The author and the reviewers validate the content of this pull request with the following checklist, in addition to the [automated tests](code_review_checklist.adoc).

- Any procedure:
  - [x] Successfully tested.
- Any page or link rename:
  - [x] The page contains a redirection for the previous URL.
  - Propagate the URL change in:
    - [x] Dashboard [default branding data](https://github.com/eclipse-che/che-dashboard/blob/main/packages/dashboard-frontend/src/services/bootstrap/branding.constant.ts)
    - [x] Chectl [constants.ts](https://github.com/che-incubator/chectl/blob/master/src/constants.ts)
- [x] Builds on [Eclipse Che hosted by Red Hat](https://workspaces.openshift.com).
- [x] the *`Validate language on files added or modified`* step reports no vale warnings.

image

### CODE REVIEW

File: modules/administration-guide/pages/upgrading-che-7-41-on-openshift.adoc [hunk 1]
```diff
@@ -0,0 +1,52 @@
+:navtitle: Upgrading {prod-short} {prod-last-version-pre-dwo} on OpenShift
+:keywords: administration-guide, migration, devworkspace
+:page-aliases: 
+
+[id="upgrading-{prod-id-short}-{prod-last-version-pre-dwo}-on-openshift_{context}"]
+= Upgrading {prod-short} {prod-last-version-pre-dwo} on OpenShift
+
+A new workspace engine, the {devworkspace}, and a new authentication system has been introduced in version {prod-first-version-using-dwo} of {prod}. Upgrading from version {prod-last-version-pre-dwo} or earlier requires to follow the migration procedure described here.
```
Comment (themr0c) [inline modules/administration-guide/pages/upgrading-che-7-41-on-openshift.adoc]:
```suggestion
The workspace engine and authentication system used until version {prod-last-version-pre-dwo} are deprecated. Upgrading from version {prod-last-version-pre-dwo} requires manual steps.
```

File: antora.yml [hunk 2]
```diff
@@ -64,15 +64,26 @@ asciidoc:
     prod-cli: chectl
     prod-deployment: che
     prod-docs-url: https://www.eclipse.org/che/docs
+    prod-docs-url-backup-recovery: link:https://www.eclipse.org/che/docs/che-7/administration-guide/backup-and-recovery/[Backup and recovery]
+    prod-docs-url-enable-oauth: link:https://www.eclipse.org/che/docs/che-7/administration-guide/configuring-openshift-oauth/[Configuring OpenShift OAuth]
+    prod-first-version-using-dwo: v7.42
```
Comment (themr0c) [inline antora.yml]:
```suggestion
```

File: antora.yml [hunk 3]
```diff
@@ -64,15 +64,26 @@ asciidoc:
...
     prod-home: /home/user
     prod-host: che-host
     prod-id-short: che
     prod-id: eclipse-che
+    prod-last-version-pre-dwo: v7.41
     prod-namespace: eclipse-che
+    prod-next-channel: next
+    prod-next-channel-catalog-source: eclipse-che-openshift-catalog-next
+    prod-next-channel-catalog-source-image: quay.io/eclipse/eclipse-che-openshift-opm-catalog:next
+    prod-next-channel-package: eclipse-che-preview-openshift
```
Comment (themr0c) [inline antora.yml]:
```suggestion
```

**Review by themr0c — APPROVED** [review]

**Comment by themr0c** [comment]
It does replace https://github.com/eclipse-che/che-docs/pull/2159, correct?

**Comment by l0rd** [comment]
> It does replace #2159, correct?

Correct. Once we merge this one I will close #2159. I am still testing the migration scripts. I will comment once I am done.

**Comment by l0rd** [comment]
I have updated the scripts but for some reason Che is unstable after the migration:
- The devworkspace controller fails to start with the following message (but that can be fixed removing the devworkspace-webhook-server deployment, mutating, and validating webhooks as described in [the doc](https://docs.openshift.com/container-platform/4.10/web_console/odc-about-web-terminal.html#deleting-the-devworkspace-operator-dependency)):
```
msg":"failed to setup webhooks","error":"conflicting webhook definition found on cluster, webhook mutate.devworkspace-controller.svc clientConfig points at namespace openshift-operators"
```
- Posgres pod fails to restart too with the following message:
```
waiting for server to start....2022-03-31 23:42:43.576 UTC [27] LOG:  redirecting log output to logging collector process
2022-03-31 23:42:43.576 UTC [27] HINT:  Future log output will appear in directory "log".
. stopped waiting
pg_ctl: could not start server
Examine the log output.
```

That may be related to the fact that I have tested the migration with some running workspaces. I will continue the investigation on Monday.

**Comment by themr0c** [comment]
OK. Merge when it's tested successfully.

**Pull request merged**